gap/PGMatrix.gi

```diff
@@ -1,56 +1,89 @@
+end );
+
+InstallMethod( GetTriangleGroup, [ IsPGMatricesOfGeneratorsObj and IsPGMatricesOfGeneratorsComponentRep ], 
+function(pgMatsGs)
+	return pgMatsGs!.fulltg;
+end );
+
+InstallMethod( GetProperTriangleGroup, [ IsPGMatricesOfGeneratorsObj and IsPGMatricesOfGeneratorsComponentRep ], 
+function(pgMatsGs)
+	return pgMatsGs!.tg;
+end );
+
+InstallMethod( TGQuotientName, [ IsPGMatricesOfGeneratorsObj and IsPGMatricesOfGeneratorsComponentRep ], 
+function(pgMatsGs)
+	return TGQuotientName(pgMatsGs!.tgquotient);
+end );
+
+InstallMethod( GetTGQuotient, [ IsPGMatricesOfGeneratorsObj and IsPGMatricesOfGeneratorsComponentRep ], 
+function(pgMatsGs)
+	return pgMatsGs!.tgquotient;
 end );
 
-InstallMethod( TGQuotientName, [ IsPGMatrixElementsObj and IsPGMatrixElementsComponentRep ], 
-function(pgMatElements)
-	return pgMatElements!.quotient;
+InstallMethod( IsSparse, [ IsPGMatricesOfGeneratorsObj and IsPGMatricesOfGeneratorsComponentRep ], 
+function(pgMatsGs)
+	return pgMatsGs!.sparse;
 end );
 
-InstallMethod( IsSparse, [ IsPGMatrixElementsObj and IsPGMatrixElementsComponentRep ], 
-function(pgMatElements)
-	return pgMatElements!.sparse;
+InstallMethod( GeneratorNames, [ IsPGMatricesOfGeneratorsObj and IsPGMatricesOfGeneratorsComponentRep ], 
+function(pgMatsGs)
+	return pgMatsGs!.GeneratorNames;
 end );
 
-InstallMethod( GetElements, [ IsPGMatrixElementsObj and IsPGMatrixElementsComponentRep ], 
-function(pgMatElements)
-	return pgMatElements!.elements;
+InstallMethod( PGMatricesRec, [ IsPGMatricesOfGeneratorsObj and IsPGMatricesOfGeneratorsComponentRep ], 
+function(pgMatsGs)
+	return pgMatsGs!.PGMatsRec;
 end );
 
 
-InstallMethod( PrintString, [ IsPGMatrixElementsObj and IsPGMatrixElementsComponentRep ], 
-function(pgMatElements)
-	return Concatenation( "PGMatrixElements( ", 
-		"signature = ",PrintString(Signature(pgMatElements)), ", ",
-		"quotient = ", PrintString(TGQuotientName(pgMatElements)), ", ",
-		"sparse = ", PrintString(IsSparse(pgMatElements)), ", ",
-		"elements = ", PrintString(GetElements(pgMatElements)), 
+InstallMethod( PrintString, [ IsPGMatricesOfGeneratorsObj and IsPGMatricesOfGeneratorsComponentRep ], 
+function(pgMatsGs)
+	return Concatenation( "PGMatricesOfGenerators( ", 
+		PrintString(GetTriangleGroup(pgMatsGs)), ", ",
+		PrintString(GetProperTriangleGroup(pgMatsGs)), ", ",
+		PrintString(GetTGQuotient(pgMatsGs)), ", ",
+		"sparse = ", PrintString(IsSparse(pgMatsGs)), ", ",
+		"GeneratorNames = ", PrintString(GeneratorNames(pgMatsGs)),  ", ",
+		"PGMatricesRec = ", PrintString(PGMatricesRec(pgMatsGs)), 
 	")" );
 end );
 
-InstallMethod( PrintObj, [ IsPGMatrixElementsObj and IsPGMatrixElementsComponentRep ], 
-function(pgMatElements)
-	Print(PrintString(pgMatElements));
+InstallMethod( PrintObj, [ IsPGMatricesOfGeneratorsObj and IsPGMatricesOfGeneratorsComponentRep ], 
+function(pgMatsGs)
+	Print(PrintString(pgMatsGs));
 end );
 
-constructTGGroups@ := function(fulltg, tg)
+
+constructTGFpAndEmb@ := function(fulltg, tg)
```

Comment:
If these can be used in other files, it would be appropriate to define them in Internal.g (or an additional file for such internal functions) such that we have all internal functions that are reused together in a (collection of) file(s).